PredictMelanomaServer.py

```diff
@@ -37,17 +53,98 @@ def test_server():
     output = {"Results": text}
+
     return jsonify(output)
 
 
 @app.route("/sample_image", methods=['POST'])
 def test_sample():
-    if request.json["currentImageString"] != '':
-        output = {"Results": "Received!"}
-    else:
-        output = {"Results": "Not received."}
+    try:
+        if request.json["currentImageString"] != '':
+            output = {"Results": "Received!"}
+        else:
+            output = {"Results": "Not received."}
+    except KeyError:
+        return send_error("Invalid key. Input dictionary.\nkey:"
+                          " currentImageString\nvalue: image encoded as base "
+                          "64 string", 400)
+    except TypeError:
+        return send_error("Invalid input. Input dictionary.\nkey:"
+                          " currentImageString\nvalue: image encoded as "
+                          "base 64 string", 400)
 
     return jsonify(output)
+
+
+@app.route("/melanoma/prediction", methods=['POST'])
+def melanoma_prediction():
+    from get_prediction import get_prediction
+    from base64 import decodestring
+    from matplotlib.image import imread
+
+    try:
+        raw_string = request.json["currentImageString"]
+    except KeyError:
+        return send_error("Invalid input. Input dictionary.\nkey: "
+                          "currentImageString\nvalue: image encoded as "
+                          "base 64 string", 400)
+    try:
+        start_index = raw_string.find(",") + 1
+        b64img_string = raw_string[start_index:]
+        imgdata = decodestring(b64img_string)
+    except TypeError:
+        return send_error("Incorrect padding: cannot decode base 64 str", 400)
+    except UnicodeDecodeError:
+        return send_error("Problem decoding input", 400)
+
+    try:
+        filename = "temp.jpg"
+        with open(filename, "wb") as image_out:
+            image_out.write(imgdata)
+        img = imread(filename)
+    except IOError:
+        return send_error("Please input base 64 encoded image", 400)
+
+    try:
+        if validate(img) is False:
+            raise AssertionError
+    except AssertionError:
+        return send_error("Wrong img size. Expected numpy.ndarray of shape "
+                          "(x, y, 3)", 500)
+    try:
+        (labels, predictions) = get_prediction(img)
+        malignant_prob = str(round(predictions[1]*100, 2)) + " %"
+        non_malignant_prob = str(round(predictions[0]*100, 2)) + " %"
+    except ValueError:
+        return send_error("get_prediction interrupted by error", 500)
+
+    if predictions[0] > predictions[1]:
+        diagnosis = "Congratulations! You do not have melanoma."
+    elif predictions[0] < predictions[1]:
+        diagnosis = "Uh oh. You should see a doctor to check out this " \
+                    "skin lesion."
+    else:
+        diagnosis = "It is too difficult to tell. Go see a doctor if " \
+                    "you are worried."
+
+    results = {"type": str(type(img)), "shape": str(img.shape),
```

Comment:
[nit] for readability, I would make each dictionary entry be its own line.  Some lines have 2 keys, some 1.